### PR TITLE
Fix ghc source plugins for GHC 9.2

### LINKED ISSUE #2989: HLS reports non-existent errors when using Calamity and Polysemy with Polysemy.Plugin (polysemy-plugin)
haskell-language-server-wrapper --probe-tools:
```
haskell-language-server version: 1.7.0.0 (GHC: 9.2.2) (PATH: /home/mon/.local/share/ghcup/hls/1.7.0.0/lib/haskell-language-server-1.7.0.0/bin/haskell-language-server-wrapper)
Tool versions found on the $PATH
cabal:          3.6.2.0
stack:          Not found
ghc:            9.2.2
```
(I launched HLS in the terminal with `haskell-language-server-wrapper` to provide the errors it reports in the paste below)
HLS reports errors that seem to not exist, because building & running the project works, and testing the program results in correct behaviour. I have this paste: https://bpa.st/J32A which has the Main.hs and the errors that HLS thinks exist. The git repository is provided later on in this issue.

### Your environment

Which OS do you use:
Arch Linux (Artix variant)
Which LSP client (editor/plugin) do you use:
doom emacs+lsp-mode
Describe your project (alternative: link to the project): https://git.sr.ht/~monaaraj/bot

### Steps to reproduce
I just run doom emacs and open up my project. This issue happens only with this calamity and polysemy (with polysemy-plugin) project, HLS acts well in every other project I have.

### Expected behaviour
The program should be deemed without error, if GHC builds and runs the program, with the program successfully running without issue.

### Actual behaviour
HLS reports very very weird errors regarding Internal.Typeable things, P.Member instance overlaps, ambiguous use of type variables... the specific errors have been provided in the bpaste link above.

### Include debug information

The HLS log seems to be too long to be pasted to any paste service, so I have provided it as an attachment file:
[hls.log](https://github.com/haskell/haskell-language-server/files/8985976/hls.log)

### PR DESCRIPTION
**Pull request by fendor**
This is supposed to fix ghc source plugins for GHC 9.4 as well, but I couldn't prove that locally.

After GHC 9.0, plugins no longer live in DynFlags, but in HscEnv. Thus, we need to make sure to pass the modified HscEnv to the subsequent typechecking phase.

Supposed to fix #2989, let's see whether it works out.

This is likely to have been introduced by #2128 (my bad!)

EDIT: ~~I am going to write a regression test for this~~ Just needed to enable the test.

<a href="https://gitpod.io/#https://github.com/haskell/haskell-language-server/pull/3311"></a>

### CODE REVIEW

File: ghcide/src/Development/IDE/Core/Compile.hs [hunk 1]
```diff
@@ -172,18 +172,18 @@ typecheckModule :: IdeDefer
 typecheckModule (IdeDefer defer) hsc tc_helpers pm = do
         let modSummary = pm_mod_summary pm
             dflags = ms_hspp_opts modSummary
-        mmodSummary' <- catchSrcErrors (hsc_dflags hsc) "typecheck (initialize plugins)"
+        (mmodSummary') <- catchSrcErrors (hsc_dflags hsc) "typecheck (initialize plugins)"
```
Comment (pepeiborra) [inline ghcide/src/Development/IDE/Core/Compile.hs:175]:
```suggestion
        mmodSummary' <- catchSrcErrors (hsc_dflags hsc) "typecheck (initialize plugins)"
```

**Review by pepeiborra — APPROVED** [review]
Good catch, thanks!

**Comment by pepeiborra** [comment]
It seems that this PR and #3309 concurrently fix the same bug, but #3309 does a bit more fixing.

**Comment by fendor** [comment]
Indeed, I did not see it! Closing this in favour of #3309

**Comment by m4dc4p** [comment]
I'm dying to have this fix and that other PR is stalled. Any chance you can merge this one?

**Comment by MorrowM** [comment]
@m4dc4p In the meantime you can `ghcup compile hls --ghc 9.2.5 -g e8f4cfee4faa3896a633560a7e056fe07a41c546`

**Pull request closed without merging**